FINE/utils.py

```diff
@@ -598,59 +596,24 @@ def output(output, verbose, val):
     if verbose == val:
         print(output)
 
-
 def checkModelclassEquality(esM, file):
-
-    def compareLists(list1,list2):
-        # Check if two given lists have the same items.
-        if len(list1) != len(list2):
-            return False
-        for item in list1:
-            if list1.count(item) != list2.count(item):
-                break
-        else: # else clause of the for-loop
-            return True
-        return False
-
     mdlList = list(esM.componentModelingDict.keys())
     compList = []
-
-    for cl in mdlList:
-        x = cl[0:len(cl)-5]
-        r = re.compile(x)
-        test = list(filter(r.match, file.sheet_names))
-        if len(test)>0:
-            compList.append(cl)
-
-    if not compareLists(mdlList, compList):
+    for sheet in file.sheet_names:
+        compList += [cl for cl in mdlList if (cl[0:-5] in sheet and cl not in compList)]
+    if len(mdlList) != len(compList):
         raise ValueError('Loaded Output does not match the given energy system model.')
-
+    for item in compList:
+        if compList.count(item) != mdlList.count(item):
+            raise ValueError('Loaded Output does not match the given energy system model.')
 
 def checkComponentsEquality(esM, file):
-
-    def checkItemLists(list1, list2):
-        # Check if a list contains the items of another list.
-        # It´s true if the longer list contains all items of the shorter list.
-        if len(list1) <= len(list2):
-            l1 = list1
-            l2 = list2
-        else:
-            l1 = list2
-            l2 = list1
-
-        for item in l1:
-            if l1.count(item) != l2.count(item):
-                break
-        else:
-            return True
-        return False
-
     components = []
     complist = list(esM.componentNames.keys())
     for mdl in esM.componentModelingDict.keys():
-        sheet = str(mdl[0:len(mdl) - 5] + 'OptSummary')
-        readSheet = pd.read_excel(file, sheet_name=sheet, index_col=[0, 1, 2, 3])
-        components = components + list(readSheet.index.levels[0])
+        readSheet = pd.read_excel(file, sheetname=mdl[0:-5] + 'OptSummary', index_col=[0, 1, 2, 3])
+        components += list(readSheet.index.levels[0])
+    for item in components:
+        if components.count(item) != complist.count(item):
+            raise ValueError('Loaded Output does not match the given energy system model.')
```

Comment:
Hi Theresa,
you can use set(components) <= set(complist) to compare the lists and avoid the loop.